### PR TITLE
Consistent test assert names for Numericals

### PR DESCRIPTION
**Pull request by javatarz**
* Rename `should equal to` to `should be equal to`
* Rename `should be equal to` to `should not be equal to`
* Rename `shouldEqualTo` to `shouldBeEqualTo`
* Rename `shouldNotEqualTo` to `shouldNotBeEqualTo`

### CODE REVIEW

**Comment by MarkusAmshove** [comment]
If we introduce such breaking changes into v2 we should also make the non-backtick assertions consistent with the (new) backtick assertions.

Maybe we can also get some feedback here from @goreRatzete, @Egorand and/or @guenhter

File: src/main/kotlin/org/amshove/kluent/CharSequenceBacktick.kt [hunk 1]
```diff
@@ -24,9 +24,9 @@ fun CharSequence.`should be blank`() = this.shouldBeBlank()
 
 fun CharSequence?.`should be null or blank`() = this.shouldBeNullOrBlank()
 
-infix fun String.`should equal to`(theOther: String) = this.shouldEqualTo(theOther)
+infix fun String.`should be equal to`(theOther: String) = this.shouldEqualTo(theOther)
```
Comment (Egorand) [inline src/main/kotlin/org/amshove/kluent/CharSequenceBacktick.kt]: I guess both backtick and camel-case versions should be changed

Comment (Egorand) [inline src/main/kotlin/org/amshove/kluent/CharSequenceBacktick.kt]: Since this is a breaking change, I suggest deprecating all occurences of `should equal to` and removing them in Kluent 2.x

File: build.gradle [hunk 2]
```diff
@@ -40,7 +40,6 @@ dependencies {
     compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
     compile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
     compile "com.nhaarman:mockito-kotlin-kt1.1:$mockito_version"
-    testCompile 'junit:junit:4.12'
```
Comment (Egorand) [inline build.gradle]: Do tests still run without the JUnit dependency?

File: docs/README.md [hunk 3]
```diff
@@ -4,85 +4,73 @@ Kluent is a "Fluent Assertions" library written specifically for Kotlin.
 
 It uses the [Infix-Notations](https://kotlinlang.org/docs/reference/functions.html#infix-notation "Infix-Notation") and [Extension Functions](https://kotlinlang.org/docs/reference/extensions.html#extension-functions "Extension Functions") of Kotlin to provide a fluent wrapper around JUnit-Asserts and Mockito.
 
- [ ![Download](https://api.bintray.com/packages/markusamshove/maven/Kluent/images/download.svg) ](https://bintray.com/markusamshove/maven/Kluent/_latestVersion) [![Build Status](https://travis-ci.org/MarkusAmshove/Kluent.svg?branch=master)](https://travis-ci.org/MarkusAmshove/Kluent)
-
+[![Download](https://api.bintray.com/packages/markusamshove/maven/Kluent/images/download.svg) ](https://bintray.com/markusamshove/maven/Kluent/_latestVersion) [![Build Status](https://travis-ci.org/MarkusAmshove/Kluent.svg?branch=master)](https://travis-ci.org/MarkusAmshove/Kluent)
```
Comment (Egorand) [inline docs/README.md]: I think it would be better if you opened a separate PR for README improvements

File: build.gradle (hunk 2 above)
Comment (javatarz) [reply build.gradle]: Yes. The tests use Spek as a test runner. The all the builds in the PR have passed.

File: src/main/kotlin/org/amshove/kluent/CharSequenceBacktick.kt (hunk 1 above)
Comment (javatarz) [reply src/main/kotlin/org/amshove/kluent/CharSequenceBacktick.kt]: Ok. Good call. I'll make those changes and update the PR :)

**Comment by javatarz** [comment]
@Egorand Thanks for the feedback. Made the changes to deprecate existing methods so the framework is backwards compatible. The project, internally, doesn't use any deprecated methods.

**Comment by javatarz** [comment]
@MarkusAmshove: This one's ready to be merged and can be v1.30. It doesn't have any breaking changes and it _fixes a bug_ in our naming scheme.

**Comment by MarkusAmshove** [comment]
Thanks 👍

**Comment by JcMinarro** [comment]
I have seen that `BasicBacktick`'s assertions are not consistent with this change, should we implement them?
I can do a PR for it if you think it is interesting.

**Comment by MarkusAmshove** [comment]
@JcMinarro I would like to have them consistent so sure, give it a shot  :-)
We should however not delete the old ones, to keep backward compatibility. 
You can annotate them as deprecated like the other assertions that got renamed (so that the ide can fix then) and them we'll remove then with 2.0
